src/ImageSharp/Formats/WebP/WebPLossyDecoder.cs

```diff
@@ -0,0 +1,1414 @@
+            vp8FilterHeader.Sharpness = (int)this.bitReader.ReadValue(3);
+            vp8FilterHeader.UseLfDelta = this.bitReader.ReadBool();
+
+            dec.Filter = (vp8FilterHeader.Level is 0) ? LoopFilter.None : vp8FilterHeader.LoopFilter;
+            if (vp8FilterHeader.UseLfDelta)
+            {
+                // Update lf-delta?
+                if (this.bitReader.ReadBool())
+                {
+                    bool hasValue;
+                    for (int i = 0; i < vp8FilterHeader.RefLfDelta.Length; i++)
+                    {
+                        hasValue = this.bitReader.ReadBool();
+                        if (hasValue)
+                        {
+                            vp8FilterHeader.RefLfDelta[i] = this.bitReader.ReadSignedValue(6);
+                        }
+                    }
+
+                    for (int i = 0; i < vp8FilterHeader.ModeLfDelta.Length; i++)
+                    {
+                        hasValue = this.bitReader.ReadBool();
+                        if (hasValue)
+                        {
+                            vp8FilterHeader.ModeLfDelta[i] = this.bitReader.ReadSignedValue(6);
+                        }
+                    }
+                }
+            }
+
+            int extraRows = WebPConstants.FilterExtraRows[(int)dec.Filter];
+            int extraY = extraRows * dec.CacheYStride;
+            int extraUv = (extraRows / 2) * dec.CacheUvStride;
+            dec.CacheYOffset = extraY;
+            dec.CacheUvOffset = extraUv;
+        }
+
+        private void ParsePartitions(Vp8Decoder dec)
+        {
+            uint size = this.bitReader.Remaining - this.bitReader.PartitionLength;
+            int startIdx = (int)this.bitReader.PartitionLength;
+            Span<byte> sz = this.bitReader.Data.Slice(startIdx);
+            int sizeLeft = (int)size;
+            dec.NumPartsMinusOne = (1 << (int)this.bitReader.ReadValue(2)) - 1;
+            int lastPart = dec.NumPartsMinusOne;
+
+            int partStart = startIdx + (lastPart * 3);
+            sizeLeft -= lastPart * 3;
+            for (int p = 0; p < lastPart; ++p)
+            {
+                int pSize = sz[0] | (sz[1] << 8) | (sz[2] << 16);
+                if (pSize > sizeLeft)
+                {
+                    pSize = sizeLeft;
+                }
+
+                dec.Vp8BitReaders[p] = new Vp8BitReader(this.bitReader.Data, (uint)pSize, partStart);
+                partStart += pSize;
+                sizeLeft -= pSize;
+                sz = sz.Slice(3);
+            }
+
+            dec.Vp8BitReaders[lastPart] = new Vp8BitReader(this.bitReader.Data, (uint)sizeLeft, partStart);
+        }
+
+        private void ParseDequantizationIndices(Vp8Decoder decoder)
+        {
+            Vp8SegmentHeader vp8SegmentHeader = decoder.SegmentHeader;
+
+            int baseQ0 = (int)this.bitReader.ReadValue(7);
+            bool hasValue = this.bitReader.ReadBool();
```

Comment:
reusing hasValue has no benefit here IMHO,
I'd remove the variable and inline the ReadBool() calls into the ternary operator lines, like:
`int dqy1DC = this.bitReader.ReadBool() ? this.bitReader.ReadSignedValue(4) : 0;`

to make this even more Readable, a local function ReadOptionalValue() is another imrovement (replacing the first suggestion)